apps/etterlatte-behandling/src/main/kotlin/behandling/klienter/Norg2Klient.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.etterlatte.behandling.klienter
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.ResponseException
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.http.isSuccess
+import kotlinx.coroutines.runBlocking
+import no.nav.etterlatte.behandling.domain.ArbeidsFordelingEnhet
+import no.nav.etterlatte.behandling.domain.ArbeidsFordelingRequest
+import no.nav.etterlatte.libs.common.logging.getXCorrelationId
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+interface Norg2Klient {
+    fun hentEnheterForOmraade(
+        tema: String,
+        omraade: String
+    ): List<ArbeidsFordelingEnhet>
+}
+
+class Norg2KlientImpl(private val client: HttpClient, private val url: String) : Norg2Klient {
+    private val logger = LoggerFactory.getLogger(Norg2KlientImpl::class.java)
+
+    private val cache = Caffeine.newBuilder().expireAfterWrite(Duration.ofMinutes(15))
+        .build<ArbeidsFordelingRequest, List<ArbeidsFordelingEnhet>>()
+
+    override fun hentEnheterForOmraade(tema: String, omraade: String): List<ArbeidsFordelingEnhet> =
+        hent(ArbeidsFordelingRequest(tema, omraade))
+
+    private fun hent(request: ArbeidsFordelingRequest): List<ArbeidsFordelingEnhet> = try {
+        val cachedInfo = cache.getIfPresent(request)
+
+        if (cachedInfo != null) {
+            logger.info("Fant cachet enheter for tema og omraade $request")
```

Comment:
Skal vi logge hele requesten?

Reply:
Er jo bare to felter - tema og område. Kan logge dem hver for seg hvis vi vil